### ISSUE TITLE
Problem with #880 Enable amp-img => img transformation for hero images

### ISSUE THREAD

**Issue by mjuzefiszyn** [ampproject/amp-toolbox#946]
Today we detected an issue with our SSR optimization process. This is very strange because it is something we didn't see before. 

Let me explain: on each article we have 2 breakpoints for the HERO image: a mobile version and a desktop version. We are manually preloading 1 of the images depending on the breakpoint. 

```html
<amp-img data-hero width="404" height="600" class="lg-hide sm-hide md-hide" layout="responsive" src="/__export/1603754286417/sites/mui/img/2020/10/26/arc_aloe_vera-hd.jpg_2108327492.jpg"  alt="Prepara tu propio jabón antiarrugas con aloe vera ¡en 3 pasos!" data-amp-auto-lightbox-disable></amp-img>

<amp-img data-hero width="1425" height="720" class="xs-hide" layout="responsive" src="/__export/1603754286417/sites/mui/img/2020/10/26/arc_aloe_vera-hd.jpg_215056869.jpg"  alt="Prepara tu propio jabón antiarrugas con aloe vera ¡en 3 pasos!" data-amp-auto-lightbox-disable></amp-img>
```

We are not using the automatic preload since it does not support responsive images. 

On the following URL you can see only 1 hero image is loaded (bypassing AMP OPTIMIZER):
[https://origen.mui.today/belleza/Prepara-tu-propio-jabon-antiarrugas-con-aloe-vera-en-3-pasos-20201026-0055.html](https://origen.mui.today/belleza/Prepara-tu-propio-jabon-antiarrugas-con-aloe-vera-en-3-pasos-20201026-0055.html
)
But on the production version with SSR the mobile image is loaded two times but most strangely, the desktop version is loaded on the mobile breakpoint. So we end up loading 3 imagenes while we want to load only 1. This is particularly painful on mobile. 

We are currently having this issues on about 30 sites . This is one example of the origin URL 
[https://origen.mui.today/belleza/Prepara-tu-propio-jabon-antiarrugas-con-aloe-vera-en-3-pasos-20201026-0055.html 
](https://origen.mui.today/belleza/Prepara-tu-propio-jabon-antiarrugas-con-aloe-vera-en-3-pasos-20201026-0055.html 
)
and this is the optimized url 
[https://mui.today/belleza/Prepara-tu-propio-jabon-antiarrugas-con-aloe-vera-en-3-pasos-20201026-0055.html.
](https://mui.today/belleza/Prepara-tu-propio-jabon-antiarrugas-con-aloe-vera-en-3-pasos-20201026-0055.html.
)
We also tested disabling our SW just in case but the error persists.

Basically the problem is when using DATA-HERO on the image attribute. Removing the data-hero attribute solves the bug. This is something we were chatting with Sebastian Benz about. 

From Sebastian:
_'_You can markup hero images using the data-hero attribute to ensure that these are rendered as fast as possible and do not depend on the AMP runtime being available. Fewer HTTPS connections on load by self-hosting the AMP runtime. _'_

That part is working, but the problem is loading the DESKTOP version on mobile and viceversa.

**Comment by sebastianbenz** [comment]
This is tricky. The correct fix would be to use the media attribute on the amp-img to declare the breakpoint. This would then get copied to the preload statement. However, this reveals a deeper problem with auto generating the preload statements. This approach simply doesn't work with responsive sites serving different versions of the same image. 

Currently, I'm leaning towards completely disabling generating preload statements. 

//cc @patrickkettner for thoughts

**Comment by sebastianbenz** [comment]
This has been fixed by #1132
